File: 00-code(源代码)/src/com/hi/dhl/algorithms/leetcode/_344/java/Solution.java
package com.hi.dhl.algorithms.leetcode._344.java;

/**
 * <pre>
 *     author: dhl
 *     date  : 2020/10/8
 *     desc  :
 * </pre>
 */

class Solution {
    public void reverseString(char[] s) {
        if (s == null || s.length <= 0) return;
        int i = 0;
        int j = s.length - 1;
        while (i <= j) {
            char tmp = s[i];
            s[i] = s[j];
            s[j] = tmp;
            i++;
            j--;
        }
    }
}

File: 00-code(源代码)/src/com/hi/dhl/algorithms/leetcode/_344/kotlin/Solution.kt
package com.hi.dhl.algorithms.leetcode._344.kotlin

/**
 * <pre>
 *     author: dhl
 *     date  : 2020/10/8
 *     desc  :
 * </pre>
 */

class Solution {
    fun reverseString(s: CharArray): Unit {
        var i = 0;
        var j = s.size - 1
        while (i < j) {
            s[i] = s[j].also { s[j] = s[i] }
            i++
            j--
        }
    }
}